### ISSUE TITLE
discussion: move repo to nodejs-tensorflow org

### ISSUE THREAD

**Issue by yorkie** [yorkie/tensorflow-nodejs#6]
Hi @JIoJIaJIu, this issue is for this project to join the org [nodejs-tensorflow](https://github.com/nodejs-tensorflow). I got that you have also implemented few many tensorflow libraries like: [nodejs-tensorflow/nodejs-tensorflow](https://github.com/nodejs-tensorflow/nodejs-tensorflow).

And I think before this project is going to be joined, we should have a clear definition how these repositories works together and unify the naming rule. That would be helpful for future community maintaining :)

**Comment by JIoJIaJIu** [comment]
Hello @yorkie, I am going now to switch on [libtensorflow_cc](https://github.com/nodejs-tensorflow/nodejs-tensorflow/issues/21).  
I think you can start to move `src/` to `src/cc/framework/` in new branch (as `libtensorflow_cc` for example)
What about naming and convention - fill free to determine the rules. But I see we have really similar notation in `cpp`
What related `javascript` - [the guys](https://github.com/nodejs-tensorflow/nodejs-tensorflow/issues/6) suggested to use `Typescript` and I found it's really useful for the project
[Here](https://github.com/nodejs-tensorflow/nodejs-tensorflow/issues/24) I created the task for deeper drilldown
Can you please joint [to slack](https://github.com/nodejs-tensorflow/nodejs-tensorflow#slack) also?
Cheers )

**Comment by zixia** [comment]
👍

**Comment by yorkie** [comment]
> I think you can start to move src/ to src/cc/framework/ in new branch (as libtensorflow_cc for example)

Nope, could we still work on this repository, because basic work has been done at this repository.

> What about naming and convention - fill free to determine the rules. But I see we have really similar notation in cpp

What I said for `naming` is the repository name and org name, like `nodejs-tensorflow` and `tensorflow-nodejs`. We could unify this in merging time.

> What related javascript - the guys suggested to use Typescript and I found it's really useful for the project

I don't think there is too guys make this suggestion, and I also think in current architecture, use JavaScript is more flexible and useful :)

> Can you please joint to slack also?

My account is yorkiefixer@gmail.com, but the team doesn't invite me that I could not join :)

Ok, basically this is what my replies to your above propose, I think still there are few things to talk about then, but it's good start, let's fix it one by one :)

**Comment by yorkie** [comment]
By the way, anyone feel free to join the discussion and build the tensorflow nodejs community together :)

**Comment by JIoJIaJIu** [comment]
> Nope, could we still work on this repository, because basic work has been done at this repository.

What is the problem? You are owner of the org, so we have the same access level. It would be better to have one shared single point of development
> What I said for naming is the repository name and org name, like nodejs-tensorflow and tensorflow-nodejs. We could unify this in merging time.

We can ask googlers how would be better to name it, I took naming [from there](https://github.com/node-tensorflow/node-tensorflow). But at the same time importing from `python` or package name in `java` has notation as `tensorflow.lang` so may be would be better to rename to `tensorflow-node/js`. Choose as you want, for me it's not big deal
> I don't think there is too guys make this suggestion, and I also think in current architecture, use JavaScript is more flexible and useful :)

`TypeScript` is similar as javascript, look at it as improved `Babel` that allows you to speed up the development and improve consistency. You may look [the video](https://channel9.msdn.com/Events/Build/2017/B8088). I was really sceptic too, but the video changed my mind. I would like to have `TypeScript` as mandatory
> My account is yorkiefixer@gmail.com, but the team doesn't invite me that I could not join :)

There is [the link](https://tensor-flow-talk-invite.herokuapp.com/) so you can register by your own

Resume:
* is needed to have `org`, I suggest mine that is yours also cause there is the issues and small community history. Also we can rename it in new notation if you choose `tensorflow-node/js`. I suggest discuss it in pm on slack if you have any doubts
* is needed to use `TypeScript` just because there is no any reason not to use it

**Comment by yorkie** [comment]
> What is the problem? You are owner of the org, so we have the same access level. It would be better to have one shared single point of development

This is not the problem of access level, it's I have implemented most works based on the architecture about js and cpp, I think moving some features from nodejs-tensorflow would be low-cost if you have no problems :)

> TypeScript is similar as javascript, look at it as improved Babel that allows you to speed up the development and improve consistency. You may look the video. I was really sceptic too, but the video changed my mind. I would like to have TypeScript as mandatory

TypeScript is such that heavily, and having a extra obstacle for a developer that wants to read. Maybe we could build a nn library or model by TypeScript.

> There is the link so you can register by your own

Joined.

**Comment by JIoJIaJIu** [comment]
I am going to do the work in the org repo in parallel then. Let stay on tune

**Comment by yorkie** [comment]
Ok, let's just close this channel :)
